File: src/server/InvestmentApp-Server/Models/Enterprise.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace InvestmentApp.Models;

public class Enterprise
{
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public Guid Id { get; set; }

    public string Name { get; set; }

    public string Address { get; set; }

    public string BankAccount { get; set; }

    public long TaxNumber { get; set; }
}

File: src/client/src/app/models/enterprise.ts
import { DataTableRow } from ".";
import { Guid } from "guid-typescript";

export interface Enterprise {
    id: Guid
    name: string;
    address: string;
    bankAccount: string;
    taxNumber: number;
}

export interface EnterpriseRow extends DataTableRow {
    id: string,
    internalId: string;
    name: string;
    address: string;
    bankAccount: string;
    taxNumber: number;
}
